src/modules/position-creator/services/position.creator.transaction.ts

```diff
@@ -75,55 +77,55 @@ export class PositionCreatorTransactionService {
                 swapRoutes[swapRoutes.length - 1],
             );
 
-        const contract = lockEpochs
-            ? await this.mxProxy.getLockedTokenPositionCreatorContract()
-            : await this.mxProxy.getPostitionCreatorContract();
-
-        let interaction: Interaction;
-
-        if (lockEpochs) {
-            interaction = contract.methodsExplicit.createPairPosFromSingleToken(
-                [
-                    new U64Value(new BigNumber(lockEpochs)),
-                    new BigUIntValue(amount0Min),
-                    new BigUIntValue(amount1Min),
-                    ...swapRouteArgs,
-                ],
-            );
-        } else {
-            interaction = contract.methodsExplicit.createLpPosFromSingleToken([
-                new AddressValue(Address.fromBech32(pairAddress)),
-                new BigUIntValue(amount0Min),
-                new BigUIntValue(amount1Min),
-                ...swapRouteArgs,
-            ]);
-        }
-
         const gasLimit =
             gasConfig.positionCreator.singleToken.liquidityPosition +
             gasConfig.pairs.addLiquidity +
             gasConfig.pairs.swapTokensFixedInput.withFeeSwap *
                 swapRoutes[0].pairs.length;
 
-        interaction = interaction
-            .withSender(Address.fromBech32(sender))
-            .withGasLimit(gasLimit)
-            .withChainID(mxConfig.chainID);
+        const transactionOptions = new TransactionOptions({
+            sender: sender,
+            chainID: mxConfig.chainID,
+            gasLimit: gasLimit,
+            function: 'createLpPosFromSingleToken',
+            arguments: [
+                new AddressValue(Address.newFromBech32(pairAddress)),
+                new BigUIntValue(amount0Min),
+                new BigUIntValue(amount1Min),
+                VariadicValue.fromItems(...swapRouteArgs),
+            ],
+        });
 
         if (payment.tokenIdentifier === mxConfig.EGLDIdentifier) {
-            interaction = interaction.withValue(new BigNumber(payment.amount));
+            transactionOptions.nativeTransferAmount = payment.amount;
         } else {
-            interaction = interaction.withSingleESDTTransfer(
-                TokenTransfer.fungibleFromBigInteger(
-                    payment.tokenIdentifier,
-                    new BigNumber(payment.amount),
-                ),
-            );
+            transactionOptions.tokenTransfers = [
+                new TokenTransfer({
+                    token: new Token({
+                        identifier: payment.tokenIdentifier,
+                    }),
+                    amount: BigInt(payment.amount),
+                }),
+            ];
         }
 
-        const transaction = interaction.buildTransaction().toPlainObject();
+        if (lockEpochs) {
+            transactionOptions.function = 'createPairPosFromSingleToken';
+            transactionOptions.arguments[0] = new U64Value(
+                new BigNumber(lockEpochs),
```

Comment:
so you construct the transaction based on the 'non lock epochs' scenario and then you only update the first argument. this can be risky, because one of the function's signature might change and there can be issues. I would suggest to extract 2 functions to make the code more readable: one function for creating transaction options for non lock epochs and one for creating transaction options for lock epochs